### DISCUSSION TITLE
Debian 10 with Cobbler 2.8.5

### DISCUSSION THREAD

**Discussion by ds-04** [cobbler/cobbler#2717] [Version specific questions]
Was about to ask for help with this, then it finally worked.

Here's the problem and a solution, though perhaps the solution can be improved?

**Problem: Can PXE Debian buster (10) Netinst... but can't get it to download the preseed file from Cobbler server (2.8.5)**

**Solution: Edit the profile so kernel options are like:**
priority=critical locale=en_GB url=http://_AN_IP_/debian/_A_Debian-10-netinst.seed_

**Question**: Is there a better way of doing this? My preseed lives outside of the cobbler webserver structure...

**Comment by ds-04** [comment]
N.B. I do plan to move to 3x at some point, also Debian 11 client installs via PXE

I'd rather install 3x on something other than Centos 8 or similar though.. think SUSE Leap 15.3 wasn't quite there when I last looked, haven't read up on 3x server on Debian yet

**Reply by SchoolGuy** [reply]
So currently you will get the best support here with any openSUSE Version (Leap 15.3 or Tumbleweed) since that is the system I am working on. However I know most folks use the EPEL repos provided by @opoplawski. Otherwise we provide Docker files to build the RPMs/DEBs for yourself. [Here](https://build.opensuse.org/project/show/home:SchoolGuy:cobbler) I do package all of the Cobbler related components on a best-effort basis for all things at the same time. Please keep in mind that we currently don't do native Debian packaging but are instead using [debbuild](https://github.com/debbuild/debbuild) for having a DEB based package.

**Reply by ds-04** [reply]
Ah thanks, perhaps my Centos 8 aversion in this case is unwarranted. I mainly run Centos7/RH7 systems. Though, the Centos 8 announcement requires a rethink, but if Cobbler 3x is geared for Centos 8, then maybe I should just run it on 8 and other systems will have another OS. But openSUSE is an option also as you say.

**Reply by SchoolGuy** [reply]
Well CentOS 7 has not all deps packaged for Cobbler 3.x.x so that would require a source installation with pip, so if a package is a must for you then it will be CentOS 8 or a derivative if you stay on RH based distros. Otherwise yes openSUSE is a way to go.

**Comment by SchoolGuy** [comment]
So since Cobblers main thing is to generate the preseed files for you through templating I don't think that there is a better way then what you described. The only thing you could do (if that is possible), that you just don't use any dynamic tags inside the pressed files and then just put them in the templates folder. Since Cobbler has nothing to generated, the output should be just the same as if you edit that URL by hand. You can then set the filename of your static file to the template field and Cobbler autogenerates the URL for you.
